### PR TITLE
feat(portex): restrict cased characters in the "Fields" key be lowercase

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

File: graviti/portex/field.py [hunk 1]
```diff
@@ -49,7 +49,15 @@ def _check_value(cls, value: PortexType) -> None:
         if not isinstance(value, PortexType):
             raise TypeError(f'The value in "{cls.__name__}" should be a PortexType')
 
+    @classmethod
+    def _check_key(cls, key: str) -> None:
```
Comment (linjiX) [inline graviti/portex/field.py:53]: what if the input key is not a string?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/field.py [hunk 2]
```diff
@@ -49,7 +49,18 @@ def _check_value(cls, value: PortexType) -> None:
...
+        try:
```
Comment (linjiX) [inline graviti/portex/field.py:54]: do we need to call `super()._check_key` here?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/field.py [hunk 3]
```diff
@@ -49,7 +49,16 @@ def _check_value(cls, value: PortexType) -> None:
...
+        super()._check_key(key)
+        if not key.islower():
+            raise ValueError(
+                f'The letters in the key of class {cls.__name__}, "{key}", must be in lowercase'
+            )
+
     def _setitem(self, key: str, value: PortexType) -> None:
+        self._check_key(key)
         self._check_value(value)
         super()._setitem(key, value)
```
Comment (linjiX) [inline graviti/portex/field.py:63]: will `super()._setitem` calls `_check_key`?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/field.py [hunk 4]
```diff
@@ -49,6 +49,14 @@ def _check_value(cls, value: PortexType) -> None:
         if not isinstance(value, PortexType):
             raise TypeError(f'The value in "{cls.__name__}" should be a PortexType')
 
+    @classmethod
+    def _check_key(cls, key: str) -> None:
+        super()._check_key(key)
+        if not key.islower():
+            raise ValueError(
+                f'The letters in the key of class {cls.__name__}, "{key}", must be in lowercase'
```
[the remark this answers is not in the document]
Comment (linjiX) [reply graviti/portex/field.py:57]: ref the docstring of `str.islower`

File: graviti/portex/field.py [hunk 5]
```diff
@@ -49,6 +50,14 @@ def _check_value(cls, value: PortexType) -> None:
         if not isinstance(value, PortexType):
             raise TypeError(f'The value in "{cls.__name__}" should be a PortexType')
 
+    @classmethod
+    def _check_key(cls, key: str) -> None:
+        super()._check_key(key)
+        if re.search("[A-Z]", key):
```
Comment (linjiX) [inline graviti/portex/field.py:56]: can we use `not str.isupper` rather than regex?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/field.py (hunk 5 above)
Comment (Lee-000) [reply graviti/portex/field.py:56]: `not str.isupper` is True when not all letters in the string are uppercase. It does not distinguish cases like `abC` and `abc`.

**Review by linjiX — APPROVED** [review]

**Pull request merged**